### PR TITLE
Add Testing

### PR DESCRIPTION
**Pull request by alessandrostockman**

### CODE REVIEW

**Comment by pseudo-rnd-thoughts** [comment]
Do you know how to setup CI testing? If not, I can get someone else to do that

Could the tests be parameterized so that new tests can be added easily and to check what cases failed easily
An example can be found [here](https://github.com/openai/gym/blob/master/tests/spaces/test_spaces.py)

To do this, may require splitting the file up. I think this is fine with each function having its own file for testing. 
This would allow us to have multiple different testing functions for each jumpy function in the file to allow testing of different cases.

For functions that do no work other than `_which_np(x).func` then these can be done in a single file (`test_jumpy.py`) 
These functions should need to do much than just to pass in arguments to check that the function runs as expected

**Comment by pseudo-rnd-thoughts** [comment]
@alessandrostockman Thanks for upgrades, they generally look good. 
Could you continue to add more tests along with a short docstring on what you are testing. 
This should help ensure that the tests reflect the idea, this is particularly helpful in the future when we no longer work on the project
We won't merge this PR until testing CI has been merged which should hopefully happen soon

File: test_requirements.txt
Comment (kir0ul) [inline test_requirements.txt:1]: I think you can remove this file as I included Pytest in the development dependencies in `setup.py`: https://github.com/Farama-Foundation/Jumpy/blob/d79e7aafffa9754a3fc499b7b36bf7d8545f6b82/setup.py#L36=

**Comment by kir0ul** [comment]
@alessandrostockman Since https://github.com/Farama-Foundation/Jumpy/pull/6 has been merged, could you rebase on the `main` branch and remove [the dummy `test_to_be_removed.py` file](https://github.com/Farama-Foundation/Jumpy/blob/main/tests/test_to_be_removed.py)?

**Comment by pseudo-rnd-thoughts** [comment]
@alessandrostockman How many more functions are there to add to this PR?

**Comment by alessandrostockman** [comment]
> @alessandrostockman How many more functions are there to add to this PR?

I'm adding something very basic for every function, today I'll push everything.

File: tests/test_jumpy.py
Comment (pseudo-rnd-thoughts) [inline tests/test_jumpy.py:488]: Is this a jax or numpy function? Is it possible to get both currently?
Could we check that the output is the expected type

File: tests/test_jumpy.py
Comment (pseudo-rnd-thoughts) [inline tests/test_jumpy.py:495]: Similar to `test_eye` is it possible to get jax and numpy outputs here?

File: tests/test_jumpy.py
Comment (pseudo-rnd-thoughts) [inline tests/test_jumpy.py:515]: repeat of `test_eyes`

File: tests/test_jumpy.py
Comment (pseudo-rnd-thoughts) [inline tests/test_jumpy.py:547]: repeat of `test_eyes`

File: tests/test_segment_sum.py
Comment (pseudo-rnd-thoughts) [inline tests/test_segment_sum.py:11]: Could we add numpy and jax example?

**Review by pseudo-rnd-thoughts — APPROVED** [review]
This looks great, thanks for all of the tests. 
I have added a couple of tests, I might be wrong as I haven't looked too deeply into jumpy implementation

File: tests/test_jumpy.py
Comment (alessandrostockman) [reply tests/test_jumpy.py:488]: The fact with those functions, if I understood correctly, is that `_which_np` gets called without any argument, resulting in always using standard numpy. So technically there would be no way to get a jax array from this... I don't know what the expected behaviour should be.

Comment (pseudo-rnd-thoughts) [reply tests/test_jumpy.py:488]: Could we use a dtype parameter where users can use a jax dtype for jax array and numpy dtype for numpy array? @jkterry1 Thoughts?

**Review by pseudo-rnd-thoughts — APPROVED** [review]
LGTM

**Pull request merged**
